lib/compress/zstd_compress.c

```diff
@@ -3614,7 +3621,7 @@ size_t ZSTD_compress_usingDict(ZSTD_CCtx* cctx,
                                int compressionLevel)
 {
     ZSTD_parameters const params = ZSTD_getParams_internal(compressionLevel, srcSize, dict ? dictSize : 0, ZSTD_cpm_noAttachDict);
-    ZSTD_CCtx_params cctxParams = ZSTD_assignParamsToCCtxParams(&cctx->requestedParams, &params);
+    ZSTD_CCtx_params cctxParams = ZSTD_assignParamsToCCtxParams(&cctx->requestedParams, &params, (compressionLevel == 0) ? ZSTD_CLEVEL_DEFAULT: compressionLevel);
```

Comment:
minor : `0` or `ZSTD_NO_CLEVEL` ?

Reply:
I was thinking `0` because the semantic isn't that no compression level is set, but its a default. I am using `ZSTD_NO_CLEVEL` only when no compression level is set.

We could add a second macro for this meaning of `0`, but I'm not quite sure what I'd name it.